jangjuseong/middlewares/auth.js

```diff
@@ -0,0 +1,18 @@
+const jwt = require('jsonwebtoken');
+const SECRET_KEY = process.env.SECRET_KEY;
+
+const validateToken = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization;
+    const decoded = jwt.verify(token, SECRET_KEY);
+    if (decoded) {
+      next();
+    }
```

Comment:
주성님! 토큰을 검증하는 함수를 너무 잘 작성해주셨네요!

여기서 decode, 즉 토큰이 복호화 된 정보가 있을 때 next() 함수를 통해 다음 함수로 넘어갈 수 있는 미들웨어를 작성해주셨는데 아래와 같이 작성해주면, 해당 미들웨어 함수 이후에 바로 유저에 대한 정보를 받을 수 있을 것 같습니다!

```suggestion
    // 토큰이 없는 경우 혹은 토큰의 정보가 잘못된 경우의 예외처리
    if ( !decoded ) {
      throw new Error("Invalid User");
    }
    
    // req 객체에 새로운 user 정보 생성
    req.user = decoded.userId // << payload에 등록한 claim key
    
    return next();
```